### ISSUE TITLE
bug: Badge.set does not work but work after creating a notification. 

### ISSUE THREAD

**Issue by mahen23** [capawesome-team/capacitor-badge#47]
**Plugin version:**

        "@capawesome/capacitor-badge": "^2.0.2",

**Platform(s):**

generic #91~18.04.1-Ubuntu SMP Fri Jul 23 13:36:29 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

**Current behavior:**

Using Badge.set does not work, but works after creating a notification

**Expected behavior:**

Badge.set should work fine. 

**Steps to reproduce:**

 Badge.clear();
    Badge.set({count: 10});
    console.log("Badge Count: ", Badge.get()); // count is 0

**Related code:**

```
 Badge.clear();
    Badge.set({count: 10});
    console.log("Badge Count: ", Badge.get()); //count is 0
```

**Other information:**

**Capacitor doctor:**

💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 4.6.2
  @capacitor/core: 4.6.2
  @capacitor/android: 4.6.2
  @capacitor/ios: 4.6.2

Installed Dependencies:

  @capacitor/cli: 4.4.0
  @capacitor/ios: 4.4.0
  @capacitor/core: 4.4.0
  @capacitor/android: 4.4.0

[success] Android looking great! 👌

```
insert the output from `npx cap doctor` here
```

**Comment by robingenz** [comment]
Can you provide a screen recording so that I better understand the problem?

**Comment by mahen23** [comment]
> Can you provide a screen recording so that I better understand the problem?

Here is a screen recording, best i could do with the little time: https://youtu.be/7vF5-MqqW20

Here is my code: 

```
  setBadge() {
    console.log("Setting badge count to: ", this.badgeCount);
    Badge.set({ count: this.badgeCount });
    console.log("Get Badge Count: ", Badge.get());
    
  }

```

**Comment by robingenz** [comment]
You need to await the promise:

```ts
 async setBadge() {
    console.log("Setting badge count to: ", this.badgeCount);
    await Badge.set({ count: this.badgeCount });
	const result = await Badge.get();
    console.log("Get Badge Count: ", result);
}
```

**Comment by felipenunesdev** [comment]
I'm facing the same problem in Android. 

I've got this reponse from plugin in Chrome console:

`{
    "callbackId": "57652480",
    "pluginId": "Badge",
    "methodName": "set",
    "options": {
        "count": 2
    }
}`

But the icon badge is not showing.

**Comment by robingenz** [comment]
@felipenunesdev The app needs to be installed as PWA.

**Comment by felipenunesdev** [comment]
@robingenz Thanks for your reply!

Actually I'm using a native application. In IOS it works like a charm, the icon badge show the correct counter, but in Android the badge is not showing.

**Comment by felipenunesdev** [comment]
@robingenz the `Badge.get()` method returns the correct counter value. Other apps I have installed on my phone, shows the icon badge correctly. So I assume that the android launch version is not the problem.

**Comment by robingenz** [comment]
@felipenunesdev Please have a look at the [documentation](https://github.com/capawesome-team/capacitor-badge#quirks):

> On Android not all launchers support badges. This plugin uses ShortcutBadger. All supported launchers are listed there.

Not all Android launcher are supported. Please have a look at [ShortcutBadger](https://github.com/leolin310148/ShortcutBadger) and check if your device is supported.

If you have more questions, please create a new issue.

**Comment by mahen23** [comment]
> You need to await the promise:
> 
> ```js
>  function async setBadge() {
>     console.log("Setting badge count to: ", this.badgeCount);
>     await Badge.set({ count: this.badgeCount });
> 	const result = await Badge.get();
>     console.log("Get Badge Count: ", result);
> }
> ```

Thanks for the reply, the code works fine, but still the count on the app icon is not getting updated. Only after a notification has been created does the badge count gets applied: 

2023-01-23 10:47:10.220 21082-21082 Capacitor/Console       com.lamaitrise.mon.espace            I  File: http://localhost/ - Line 228 - Msg: undefined
2023-01-23 10:47:10.518 21082-21082 Capacitor/Console       com.lamaitrise.mon.espace            I  File: http://localhost/ - Line 228 - Msg: undefined
2023-01-23 10:47:11.146 21082-21082 Capacitor/Console       com.lamaitrise.mon.espace            I  File: http://localhost/8152.js - Line 1 - Msg: Setting badge count to:  3
2023-01-23 10:47:11.206 21082-21082 Capacitor/Console       com.lamaitrise.mon.espace            I  File: http://localhost/ - Line 228 - Msg: undefined
2023-01-23 10:47:11.228 21082-21082 Capacitor/Console       com.lamaitrise.mon.espace            I  File: http://localhost/8152.js - Line 1 - Msg: Get Badge Count:  {"count":3}
2023-01-23 10:47:13.693 21082-21082 ViewRootIm...nActivity] com.lamaitrise.mon.espace            I  MSG_WINDOW_FOCUS_CHANGED 0 1
2023-01-23 10:48:12.161 21082-21082 Capacitor/Console       com.lamaitrise.mon.espace            I  File: http://localhost/8152.js - Line 1 - Msg: After Creating notification
2023-01-23 10:48:12.165 21082-21082 Capacitor/Console       com.lamaitrise.mon.espace            I  File: http://localhost/8152.js - Line 1 - Msg: Setting badge count to:  3
2023-01-23 10:48:12.273 21082-21082 Capacitor/Console       com.lamaitrise.mon.espace            I  File: http://localhost/ - Line 228 - Msg: undefined
2023-01-23 10:48:12.288 21082-21082 Capacitor/Console       com.lamaitrise.mon.espace            I  File: http://localhost/ - Line 228 - Msg: undefined
2023-01-23 10:48:12.311 21082-21082 Capacitor/Console       com.lamaitrise.mon.espace            I  File: http://localhost/8152.js - Line 1 - Msg: Get Badge Count:  {"count":3}

**Comment by robingenz** [comment]
@mahen23 Please provide a Minimal, Reproducible Example (see [How to create a Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example)) so I can debug the issue.

**Comment by mahen23** [comment]
[badgeBug.tar.gz](https://github.com/capawesome-team/capacitor-badge/files/10479981/badgeBug.tar.gz)


> @mahen23 Please provide a Minimal, Reproducible Example (see [How to create a Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example)) so I can debug the issue.

**Comment by mahen23** [comment]
> @mahen23 Can you please publish the example as GitHub repo?

Sorry I do not know how to do that... I can google it and see how its done, but will take me lots of time.

**Comment by robingenz** [comment]
Just create a normal GitHub repository and push all files:

```
git init
git add .
git commit -m "first commit"
git branch -M main
git remote add origin https://github.com/mahen23/YOUR-REPO.git
git push -u origin main
```

**Comment by mahen23** [comment]
> 



> Just create a normal GitHub repository and push all files:
> 
> ```
> git init
> git add .
> git commit -m "first commit"
> git branch -M main
> git remote add origin https://github.com/mahen23/YOUR-REPO.git
> git push -u origin main
> ```

remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/mahen23/badgeBug.git/'

**Comment by robingenz** [comment]
> > 
> 
> > Just create a normal GitHub repository and push all files:
> > ```
> > git init
> > git add .
> > git commit -m "first commit"
> > git branch -M main
> > git remote add origin https://github.com/mahen23/YOUR-REPO.git
> > git push -u origin main
> > ```
> 
> remote: Support for password authentication was removed on August 13, 2021. remote: Please see [docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls](https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls) for information on currently recommended modes of authentication. fatal: Authentication failed for '[mahen23/badgeBug.git](https://github.com/mahen23/badgeBug.git/)'

See https://stackoverflow.com/a/68781050/6731412

**Comment by mahen23** [comment]
> > > 
> > 
> > 
> > > Just create a normal GitHub repository and push all files:
> > > ```
> > > git init
> > > git add .
> > > git commit -m "first commit"
> > > git branch -M main
> > > git remote add origin https://github.com/mahen23/YOUR-REPO.git
> > > git push -u origin main
> > > ```
> > 
> > 
> > remote: Support for password authentication was removed on August 13, 2021. remote: Please see [docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls](https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls) for information on currently recommended modes of authentication. fatal: Authentication failed for '[mahen23/badgeBug.git](https://github.com/mahen23/badgeBug.git/)'
> 
> See https://stackoverflow.com/a/68781050/6731412

Thanks for the walkthrough mate, here is my repository: https://github.com/mahen23/badgeBug/blob/main/src/app/home/home.page.ts

**Comment by robingenz** [comment]
You should only pass integers as count.
`getRandomArbitrary` also returns comma values:

```
$ Math.random() * (99 - 10) + 10
31.30685279633496 
```

Besides that, please let me know if this demo works for you: https://capacitor-plugin-demo-robingenz.vercel.app/badge

**Comment by mahen23** [comment]
> You should only pass integers as count. `getRandomArbitrary` also returns comma values:
> 
> ```
> $ Math.random() * (99 - 10) + 10
> 31.30685279633496 
> ```
> 
> Besides that, please let me know if this demo works for you: https://capacitor-plugin-demo-robingenz.vercel.app/badge

Hello, 

It makes not sense to test a demo link since the issue i'm having is when i build an APK. As per your instructions i've created a git repository to as a proof of concept that the app icon badge count is not updating. 

I cannot test the demo link you sent me since it will work inside a browser. Since my bug NEEDS REPRODUCTION, you need to compile an APK on your test machine and test the badge update count if it gets updated properly on your phone. 

I've made changes to random number generator as per your instructions.

**Comment by robingenz** [comment]
> It makes not sense to test a demo link since the issue i'm having is when i build an APK.

This is the first time you have referred to Android. You have specified
`generic #91~18.04.1-Ubuntu SMP Fri Jul 23 13:36:29 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux` as the platform related to the bug.
For this reason, I assumed that the problem occurs in the browser.

I will take another look soon.
